base.py

```diff
@@ -1,11 +1,55 @@
-from abc import ABC, abstractmethod
+from abc import ABC
+
+import joblib
+from joblib import Parallel, delayed
 from typing import Union
 
 import numpy as np
 import pandas as pd
 
+from featurizers.descriptors import fp_descriptors
+
 
 class FingerprintTransformer(ABC):
-    @abstractmethod
+    def __init__(self, n_jobs):
+        if n_jobs == -1:
+            self.n_jobs = joblib.cpu_count()
+        else:
+            self.n_jobs = n_jobs
+
+        self.fp_args = None
+        self.fp_descriptor_name: str = ""
+
+    def _transform(self, X: Union[pd.DataFrame, np.ndarray]) -> np.ndarray:
+        """
+        Helper function to be executed in each sub-process.
+
+        :param X: subset of original X data
+        :return: np.array containing calculated fingerprints for each molecule
+        """
+        result = np.array(
+            [
+                fp_descriptors[self.fp_descriptor_name](x, **self.fp_args)
+                for x in X
+            ]
+        )
+        return result
+
     def transform(self, X: Union[pd.DataFrame, np.ndarray]) -> np.ndarray:
```

Comment:
Trzeba będzie coś zrobić z walidacją wejścia. Generalnie w tym wypadku to mocno przypomina CountVectorizer. Wchodzi lista "dziwnych" obiektów, wychodzi ładna macierz. Tutaj wchodzą albo stringi (SMILES), albo molekuły RDKit. Trzeba zrobić:
1. Sprawdzenie typu
2. Zamianę na molekuły, jeżeli trzeba
3. Zmienić przyjmowany typ, to może być tak z grubsza `Iterable[str | Mol]`